torch/nn/utils/clip_grad.py

```diff
@@ -31,3 +31,20 @@ def clip_grad_norm(parameters, max_norm, norm_type=2):
         for p in parameters:
             p.grad.data.mul_(clip_coef)
     return total_norm
+
+
+def clip_grad_value(parameters, clip_value):
+    r"""Clips gradient of an iterable of parameters at specified value.
+
+    Gradients are modified in-place.
+
+    Arguments:
+        parameters (Iterable[Tensor]): an iterable of Tensors that will have
+            gradients normalized
+        clip_value (float or int): maximum allowed value of the gradients
+            The gradients are clipped in the range [-clip_value, clip_value]
+    """
+    parameters = list(filter(lambda p: p.grad is not None, parameters))
```

Comment:
It’s better to avoid materializing the list. You can put the filter directly as the for loop iterator

Reply:
Ooops forgot to commit my test, sorry!